### PR TITLE
Permissions

### PR DESCRIPTION
**Pull request by topless**

### CODE REVIEW

**Comment by topless** [comment]
Hey @ntarocco and @slint can you have a look on this? By the way you can't hurt my feelings. Cheers!

File: invenio_record_editor/utils.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2018 CERN.
+#
+# Invenio-Record-Editor is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Invenio module serving a generic record editor."""
```
Comment (ntarocco) [inline invenio_record_editor/utils.py]: Better comment?

File: invenio_record_editor/views.py [hunk 2]
```diff
@@ -7,28 +7,27 @@
...
     template_folder='templates',
     static_folder='static',
 )
 
 
 @blueprint.route('/')
+@need_editor_permissions('editor-view')
 def index():
-    """Render a basic view."""
+    """Render a basic view, with dummy permission editor-view."""
```
Comment (ntarocco) [inline invenio_record_editor/views.py]: Well, it is not dummy normally :)

`Render the view to display the record editor` or similar?

File: invenio_record_editor/views.py [hunk 3]
```diff
@@ -7,28 +7,27 @@
 
 """Invenio module serving a generic record editor."""
```
Comment (ntarocco) [inline invenio_record_editor/views.py]: Comment?

File: setup.py [hunk 4]
```diff
@@ -37,11 +37,15 @@
...
 
 setup_requires = [
+    'Babel>=1.3',
     'pytest-runner>=2.6.2',
 ]
 
 install_requires = [
+    'Flask-BabelEx>=0.9.2',
     'invenio-assets>=1.0.0',
+    'invenio-db>=1.0.2',
```
Comment (ntarocco) [inline setup.py]: @Alex (cannot tag you), here we should require db with the optional sqlite, etc... actually, db is not required for this module, right?

**Review by ntarocco — COMMENTED** [review]
In general, check the doc strings around.
Also, build the documentation, fire up the index.html and see how it looks like, it might be that you have to improve it.

File: setup.py (hunk 4 above)
Comment (slint) [reply setup.py]: If it's only used in tests/example-app it should go in `tests_require` (or `example/requirements.txt`) like e.g. `invenio-records-rest` does: https://github.com/inveniosoftware/invenio-records-rest/blob/788443596aad1d86a8a06b17f7625692a30e6092/setup.py#L23

File: setup.py [hunk 5]
```diff
@@ -37,11 +37,15 @@
...
+    'invenio-access>=1.0.0',
```
Comment (slint) [inline setup.py]: I'm about to say that you might actually not need `invenio-access` at all... The `AllowAll` permission is not depending on anything from `invenio_access`

**Review by slint — APPROVED** [review]
LGTM :shipit:

File: setup.py (hunk 5 above)
Comment (topless) [reply setup.py]: Exactly my thought, now that it works end to end, I 'll start killing dependencies.

File: setup.py (hunk 4 above)
Comment (ntarocco) [reply setup.py]: @topless can u fix this?

**Comment by ntarocco** [comment]
You can fork https://github.com/inveniosoftware/invenio-records-editor and then create a PR.

File: setup.py (hunk 4 above)
Comment (topless) [reply setup.py]: @ntarocco at the most recent commit I address the aforementioned issues. Tests and example app are running and the only dependency left is `accounts` for the `jwt` token.

**Pull request merged**